### DISCUSSION TITLE
'create' is deprecated

### DISCUSSION THREAD

**Discussion by Ian-Balijawa** [pmndrs/zustand#1548] [Q&A] [answered]
'create' is deprecated.ts(6385) react.d.ts(24, 4): The declaration was marked as deprecated here.

What's the new and recommended way of creating a store since this one is being render deprecated.

**Comment by dai-shi** [comment]
New one is named export. Doesn't it show as deprecation message.
Would you please share your screenshot? What's your editor/IDE?

**Reply by Ian-Balijawa** [reply]
![Screenshot (63)](https://user-images.githubusercontent.com/49788350/213208639-7665322b-c1a1-491d-8bb7-d96312fba90f.png)

"zustand": "^4.3.2"

**Reply by Ian-Balijawa** [reply]
Please update it in the docs at https://docs.pmnd.rs/zustand/getting-started/introduction to reflect the named export way.
The docs still use the default export.
check here https://docs.pmnd.rs/zustand/getting-started/introduction
and here https://zustand-demo.pmnd.rs/

**Reply by dai-shi** [reply]
Good point. I'll fix the demo. The docs are still under construction. Docs team is working on it. Feel free to open a PR, if you like too.

I wasn't aware that the deprecation message isn't shown in the editor.
Expected to see the message like this: https://www.youtube.com/watch?v=8dkkyCwbUFM

**Reply by dai-shi** [reply, marked as answer]
#1554 fixed https://zustand-demo.pmnd.rs/

**Comment by fungw** [comment]
As suggest by dai-shi

Named export like so `import { create } from "zustand"`

**Reply by JuanmaMenendez** [reply]
Summarizing, what was deprecated was the way of importing the method.

//WRONG

`import create from 'zustand';`

//OK

`import { create } from 'zustand';`

**Comment by IntuzParthD** [comment]


**Reply by dai-shi** [reply]
That should be correct. What's your bundler?

**Comment by hendra-foo** [comment]
I'm using next.js v12.2.0 with react v18.2.0 and zustand v4.4.7, 
imported correctly like this:
`import { create } from "zustand";`
`import { createJSONStorage, persist } from "zustand/middleware";`

but still receiving warning below


> event - compiled client and server successfully in 7.1s (13278 modules)
[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'.
[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`.
[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.
[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'.
[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`.
[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.
[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'.
[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`.
[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.
[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'.
[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`.
[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.
[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'.
[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`.
[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.
[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'.
[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`.
[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.
[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'.
[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`.
[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.

**Reply by hendra-foo** [reply]
![image](https://github.com/pmndrs/zustand/assets/28421317/f3c41f69-e014-4a07-b537-2768fba0a1c1)

warning gone if I don't render my `PlateRenderer` component, but I'm just curious how come zustand warning triggered by another third party package `@udecode/plate`?

**Reply by dbritto-dev** [reply]
@hendra-foo seems that you have two versions of zustand you can confirm that by looking at your package lock file, you can fix that using `overrides` section in your package file

**Reply by zbeyens** [reply]
@hendra-foo This has been patched in Plate 27
